### DISCUSSION TITLE
Is it possible to view the Dockerfile used to build a container?

### DISCUSSION THREAD

**Discussion by sliekens** [devcontainers/community#120] [Q&A] [answered]
I'm having some issues configuring NODE_EXTRA_CA_CERTS which is needed for VSCode to be able to reach the internet from behind my corporate firewall which decrypts encrypted traffing and re-encrypts with our inhouse CA certificate.

I am setting `"containerEnv": { "NODE_EXTRA_CA_CERTS": "path-to-bundle.crt" }` inside a devcontainer feature but I guess it's not doing what I expected it to, extensions listed in `devcontainer.json` are not installed when the container starts.

Can I somehow view the Dockerfile which is generated from a devcontainer.json definition?

**Comment by samruddhikhandale** [comment]
Reading through the CLI code, it looks like we save the Dockerfile created by Features in a `/tmp` folder. However, I don't think that's available later.

See 👇 

<img width="799" alt="image" src="https://github.com/devcontainers/community/assets/24955913/5a13f252-867f-43fc-a243-659cdc05b9c5">

Looping in @joshspicer in case I am misinterpreting.

**Reply by joshspicer** [reply, marked as answer]
Yea, we do write the Dockerfile to a temporary path (and use _that_ Dockerfile when running the build).  Searching your disk for some string like `dev_containers_target_stage` as shown in @samruddhikhandale 's message above is probably the easiest way to dig through that file.  Note it's temporary and might be deleted by your OS.

That said, there's a good amount of info captured in some older, closed GitHub issues - like in [this issue](https://github.com/microsoft/vscode-remote-release/issues/6092#issuecomment-1935461173).  That might be a more helpful place to look?

If you'd like to audit the change in the CLI where `NODE_EXTRA_CA_CERTS` is handled, it was added in [this release](https://github.com/devcontainers/cli/blob/10b0ec54dcaafb943ff9165b284c95119e39e520/CHANGELOG.md?plain=1#L117)

**Reply by sliekens** [reply]
Thanks, I found the file in ` /tmp/devcontainercli-<user>/container-features/0.56.2-1710936344096/Dockerfile-with-features`

Also thank you for the link to the `NODE_EXTRA_CA_CERTS` changes. I am setting `ENV NODE_EXTRA_CA_CERTS="/usr/local/share/ca-certificates/bundle.crt"` via `containerEnv`, but maybe that is no longer needed then, I will have to investigate further.

However, I found that my real issue seems to be the same as this: https://github.com/microsoft/vscode/issues/207813

Anyhow, thanks for your time.
